content/concepts/discovery/mDNS.md

```diff
@@ -0,0 +1,51 @@
+---
+title: "MDNS"
+description: "MDNS uses a multicast system of DNS records over a local network to enable peer discovery."
+weight: 224
+---
+
+## What is mDNS?
+
+MDNS, or multicast Domain Name System, uses a multicast system of DNS records over a
+local network. Nodes broadcast topics they're interested in into the network instead of
+querying a central name server. The discovery is limited to the peers in the local network.
+
+MDNS uses UDP to enable hostname resolution on local networks, allowing it to be used
+on networks that support multicast, such as an office. In turn, it can also use DNS to
+resolve hostnames to IP addresses on the Internet. MDNS does not require a central DNS
+server to operate, as a node can resolve hostnames to IP addresses in a decentralized
+manner.
+
+## MDNS in libp2p
+
+In libp2p, mDNS can be used as part of the [rendezvous protocol](rendezvous) to enable peers
+to discover and connect. When a peer wants to connect to another peer in a libp2p network,
+it can use mDNS to resolve the hostname of the other peer to its multiaddress.
+
+MDNS follows a request-response model, where a peer broadcasts a query request to *find all peers*
+on a local network to receive DNS response messages from peers which contain the peer information
+of discovered peers. The response message is in the form of DNS record:
+
+`<service-name> PTR <peer-name>.<service-name>`,
+
+where `<service-name>` is the name of the service that is being advertised, and
+`<peer-name>` is the name of the peer [that is providing the service].
```

Comment:
It's not. We don't have any use for that field, so we just fill it with a string with random characters.